framework/util/page_guard_manager.cpp

```diff
@@ -230,15 +233,102 @@ PageGuardManager::~PageGuardManager()
     }
 }
 
+#if !defined(WIN32)
+constexpr uint32_t PageGuardManager::signal_handler_watcher_max_restores;
+uint32_t           PageGuardManager::signal_handler_watcher_restores = 0;
+
+void PageGuardManager::MarkAllTrackedMemoryAsDirty()
+{
+    std::lock_guard<std::mutex> lock(tracked_memory_lock_);
+
+    for (auto entry = memory_info_.begin(); entry != memory_info_.end(); ++entry)
+    {
+        auto memory_info = &entry->second;
+
+        memory_info->is_modified = true;
+        memory_info->status_tracker.SetAllBlocksActiveWrite();
+        SetMemoryProtection(memory_info->shadow_memory, memory_info->shadow_range, kGuardNoProtect);
+    }
+}
+
+bool PageGuardManager::CheckSignalHandler()
+{
+    assert(instance_);
+
+    instance_->signal_handler_lock_.lock();
+
+    if (instance_->exception_handler_)
+    {
+        assert(instance_->exception_handler_count_);
+
+        struct sigaction current_handler;
+        int              result = sigaction(SIGSEGV, nullptr, &current_handler);
+
+        if (result != -1)
+        {
+            if (current_handler.sa_sigaction != PageGuardExceptionHandler)
+            {
+                GFXRECON_LOG_WARNING("PageGuardManager: Signal handler has been removed. Re-installing.")
+
+                instance_->exception_handler_       = nullptr;
+                instance_->exception_handler_count_ = 0;
+                instance_->signal_handler_lock_.unlock();
+
+                instance_->AddExceptionHandler();
+
+                // Potentially we lost information. Assume everything is dirty
+                instance_->MarkAllTrackedMemoryAsDirty();
+
+                return true;
+            }
+        }
+        else
+        {
+            GFXRECON_LOG_ERROR("sigaction failed (%s)", strerror(errno));
+        }
+    }
+
+    instance_->signal_handler_lock_.unlock();
+
+    return false;
+}
+
+void* PageGuardManager::SignalHandlerWatcher(void* args)
+{
+    while (instance_->enable_signal_handler_watcher_ &&
+           signal_handler_watcher_restores < signal_handler_watcher_max_restores)
+    {
+        signal_handler_watcher_restores += CheckSignalHandler();
```

Comment:
I know `true` will evaluate to `1` here, but for clarity if `signal_handler_watcher_restores` is a `uint32_t` then I'd prefer `CheckSignalHandler` to return a `uint32_t` or do `? 1 : 0` here.

Reply:
For my own learning of the house style, would the most explicit option also be the most preferred? Like this:
```cpp
if(true == CheckSignalHandler())
{
   ++signal_handler_watcher_restores;
}
```

Reply:
That would work, too, but I don't feel like comparing a value with `true` is required.